src/CoreWCF.BuildTools/src/Extensions.cs

```diff
@@ -46,6 +46,22 @@ internal static class ParameterSymbolExtensions
     {
         public static bool IsMatchingParameter(this IParameterSymbol symbol, IParameterSymbol parameterSymbol)
             => SymbolEqualityComparer.Default.Equals(symbol.Type, parameterSymbol.Type);
+
+        public static string? GetInjectedPropertyNameValue(this IParameterSymbol symbol, INamedTypeSymbol? coreWcfInjectedAttributeSymbol)
+        {
+            foreach (AttributeData attributeData in symbol.GetAttributes())
+            {
+                if (SymbolEqualityComparer.Default.Equals(attributeData.AttributeClass, coreWcfInjectedAttributeSymbol))
+                {
+                    if (attributeData.NamedArguments.Length > 0 && attributeData.NamedArguments[0].Key == "PropertyName")
+                    {
```

Comment:
This will be fragile if we add any more properties to the attribute. Maybe use something like:
```c#
var propNameArg = attributeData.NamedArguments.Where(na => na.Key == "PropertyName").FirstOrDefault();
if (propNameArg != default)
{
    return propNameArg.Value!.ToString();
}
```
What happens if someone sets the PropertyName to null? Does it still show up in the NamedArguments? Will `Value!.ToString()` throw a NRE? What type is the Value? Is it the type of the actual property, ie a string? If so, would a cast be better? That way you would return a null. Do we want to differentiate the difference between not specifying the PropertyName vs setting it to null? If they set it to null, my thinking is that should yield a compilation error.

Reply:
I see you did something along these lines in OperationParameterInjectionGenerator.EmitOperationContract